integration/aggregate_compat_test.go

```diff
@@ -28,9 +28,14 @@ import (
 
 // aggregateStagesCompatTestCase describes aggregation stages compatibility test case.
 type aggregateStagesCompatTestCase struct {
-	skip       string                   // skip test for all handlers, must have issue number mentioned
-	pipeline   bson.A                   // required, unspecified $sort appends bson.D{{"$sort", bson.D{{"_id", 1}}}}
-	resultType compatTestCaseResultType // defaults to nonEmptyResult
+	skip           string                   // skip test for all handlers, must have issue number mentioned
+	pipeline       bson.A                   // required
+	resultType     compatTestCaseResultType // defaults to nonEmptyResult
+	resultPushdown bool                     // defaults to false
+	// when noSort is false, it appends bson.D{{"$sort", bson.D{{"_id", 1}}}}
+	// to pipeline, only if $sort stage does not exist.
+	// This ensures compat and target has the same order of results.
+	noSort bool
```

Comment:
Let's better say when `noSort` should be true, it's already false by default, but as a developer when do I need to set it true?